CRT.c

```diff
@@ -96,6 +96,9 @@ const char* const* CRT_treeStr = CRT_treeStrAscii;
 
 static const Settings* CRT_settings;
 
+static volatile sig_atomic_t CRT_terminateRequested = 0;
+static volatile sig_atomic_t CRT_terminateSignal = 0;
```

Comment:
I have a small idea that I'm not sure you would experiment on.

The `CRT_terminateSignal` variable and `CRT_terminateRequested` flag may be merged to one variable. Since the signal number would be likely zero if there's no signal received.